### PR TITLE
implement ilKioskModeService

### PR DESCRIPTION
**Pull request by nhaagen**

### CODE REVIEW

File: Services/Init/classes/class.ilInitialisation.php [hunk 1]
```diff
@@ -1978,4 +1980,17 @@ private static function initInjector(\ILIAS\DI\Container $c) {
...
 		};
 	}
+
+
+	private static function initKioskMode(\ILIAS\DI\Container $c) {
+		$c["service.kiosk_mode"] = function ($c) {
+			return new ilKioskModeService(
+				$c->ctrl(),
+				$c->language(),
+				$c->access(),
```
Comment (klees) [inline Services/Init/classes/class.ilInitialisation.php]: Please use array-access here. The methods on the `DIC` will go away at some point.

File: Services/KioskMode/classes/class.ilKioskModeService.php [hunk 2]
```diff
@@ -5,12 +5,42 @@
  * Central entry point for users of the service.
  */
 final class ilKioskModeService {
+
+	protected $ctrl;
+	protected $language;
+	protected $access;
+	protected $obj_definition;
```
Comment (klees) [inline Services/KioskMode/classes/class.ilKioskModeService.php:27]: Please add typehints here.

File: Services/KioskMode/classes/class.ilKioskModeService.php [hunk 3]
```diff
@@ -5,12 +5,42 @@
...
+
+	public function __construct(
+		ilCtrl $ctrl,
+		illanguage $language,
```
Comment (klees) [inline Services/KioskMode/classes/class.ilKioskModeService.php]: `illanguage` -> `ilLanguage`

File: Services/KioskMode/classes/class.ilKioskModeService.php [hunk 4]
```diff
@@ -19,5 +49,22 @@ public function getViewFor(\ilObject $object) {
...
+	}
+
+	/**
+	 * @return string[] with [0] => path and [1] => class name
+	 */
+	protected function getClassLocationForType(string $object_type): array
+	{
+		$class = $this->obj_definition->getClassName($object_type);
+		$location = $this->obj_definition->getLocation($object_type);
+		$full_class = "il".$class."KioskModeView";
```
Comment (klees) [inline Services/KioskMode/classes/class.ilKioskModeService.php:77]: Is it correct that `$class` *does not* contain an `il`-Prefix here?

File: Services/KioskMode/classes/class.ilKioskModeService.php [hunk 5]
```diff
@@ -19,5 +49,22 @@ public function getViewFor(\ilObject $object) {
 	 * @param	string	$object_type	needs to be a valid object type
 	 */
 	public function hasKioskMode(string $object_type) : bool {
+		list($location, $class_name) = $this->getClassLocationForType($object_type);
+		$class_path = sprintf(
+			'%s/../../../%s/class.%s.php',
+			__DIR__, $location, $class_name
+		);
+		return file_exists($class_path);
```
Comment (klees) [inline Services/KioskMode/classes/class.ilKioskModeService.php]: I think you could just check if `class_exists` and don't need the location. The location should already be resolved via the autoloader anyway.

**Review by klees — CHANGES_REQUESTED** [review]
LGTM in general, but some nitpicking.

Please be so kind and have a quick look into the README.md as well, adjust the TOC and spend a minute or two thinking about if some information is missing in there. Thx!

**Comment by klees** [comment]
Hi @smeyer-ilias, we'll make a rather standard addition to `ilInitialisation`here. Is that fine  with you?

File: Services/KioskMode/classes/class.ilKioskModeService.php [hunk 6]
```diff
@@ -19,5 +64,18 @@ public function getViewFor(\ilObject $object) {
...
+		return class_exists($class_name);
+	}
+
+	/**
+	 * @return string[] with [0] => path and [1] => class name
+	 */
+	protected function getClassLocationForType(string $object_type): array
+	{
+		$class = $this->obj_definition->getClassName($object_type);
+		$location = $this->obj_definition->getLocation($object_type);
```
Comment (klees) [inline Services/KioskMode/classes/class.ilKioskModeService.php]: We don't need `$location` anymore.

**Comment by klees** [comment]
@smeyer-ilias Any objections to the addition to `ilInitialisation`?

**Comment by klees** [comment]
@smeyer-ilias Could you have a quick look into this? There's nothing really interesting, I promise.

**Comment by smeyer-ilias** [comment]
@klees : Yes, LGTM. You can merge directly.

**Pull request merged**